src/controllers/CategoryFacetQueryController.ts

```diff
@@ -1,85 +1,65 @@
 import { CategoryFacet } from '../ui/CategoryFacet/CategoryFacet';
+import { Assert } from '../misc/Assert';
 import { QueryBuilder } from '../ui/Base/QueryBuilder';
-import { ICategoryFacetRequest } from '../rest/CategoryFacetRequest';
-import { IGroupByRequest } from '../rest/GroupByRequest';
-import { AllowedValuesPatternType } from '../rest/AllowedValuesPatternType';
-import { IGroupByValue } from '../rest/GroupByValue';
-import { Utils } from '../utils/Utils';
+import { IFacetRequest } from '../rest/Facet/FacetRequest';
+import { QueryEvents } from '../events/QueryEvents';
 
+// IDEA: maybe this could simply extend the DynamicFacetQueryController with only a few modifications
```

Comment:
I'm generally very much against inheritance... 

From my personal experience, it always end up being more pain to maintain in the long run because over time, the base class and inherited one tend to "drift" away from one another and then you end up having to create hacks to make them work, and you end up overriding methods, which always sucks.

Try to go for composition if you want to share stuff between the two classes, it's generally much easier to change in the future if we need to.